### ISSUE TITLE
Split utils into independent repository?

### ISSUE THREAD

**Issue by jerryz123** [chipsalliance/rocket-chip#3596]
Many utilities in rocketchip.utils only depend on chisel. Other projects may want to use these utilities without pulling in all of rocket-chip. These implementations are also stable with stable, simple APIs, so doing packaged releases of these independent of RC would help some users greatly.

**Type of issue**: feature request

**Development Phase**: proposal

**Comment by jerryz123** [comment]
@lordspacehog @sequencer

**Comment by lordspacehog** [comment]
yeah, i think this is a good idea. I think it'd also be great to identify other parts to split out as well. The AMBA stuff would be pretty useful on it's own as a diplomatic interface implementation for example.

**Comment by lordspacehog** [comment]
@jerryz123  @sequencer  Im starting work on this now. I opened an issue on the chisel repo to get input from them before i start sending PRs to them to upstream things from our utils. it would be great to get y'alls input on that ticket as well.

**Comment by sequencer** [comment]
You can split it into a new `build.sc` build target for now, and I'll take a look at them one by one to propose a concrete plan for deprecating or upstreaming to chisel stdlib.
